### ISSUE TITLE
Could you explain a little bit about the statistic metrics?

### ISSUE THREAD

**Issue by Jing25** [xinntao/BasicSR#35]
This is a beginner question.. In the tensorboard, there are D_fake, D_real, l_d_fake, l_d_real, l_g_fea, l_g_gan, l_g_pix. Could you please explain what they are and what the expected trend each of them should have? 

Thank you so much.

**Comment by Jing25** [comment]
How can I test SRGAN with your code? In the test_esrgan.json file, I changed "model" into "srgan" and "pretrain_model_G" into "SRGAN_bicx4_303_505.pth". But there's error.

**Comment by xinntao** [comment]
Hi @Jing25 
D_fake: The mean of D outputs for fake ones (generated SR images)
D_real: The mean of D outputs for real ones (GT)
l_d_fake: D loss for fake ones (generated SR images)
l_d_real: D loss for real ones (GT)
l_g_fea:  G loss (feature loss, e.g., percetual loss)
l_g_gan: G loss (adversarial loss )
l_g_pix: G loss (pixel loss)
Run the codes, and get their values. 

You may also need to change 
`"which_model_G": "RRDB_net" // RRDB_net | sr_resnet`
in the ` test_esrgan.json`.

**Comment by Jing25** [comment]
If D_fake/D_real is the mean of D outputs fake/real, then I guess the expected optimal value of each of them should be 0.5?
I think G loss (l_g_fea, l_g_gan and l_g_pix) should decrease with number of iteration increasing. How about D loss (l_d_fake and l_d_real)? 

Yeah. I change "which_model_G" to "sr_resnet" and then "nb" to 16. It works. But I'm just curious why I cannot change "norm_type": null to "batch"? SRGAN uses batch normalization. But why here if I change "norm_type" to "batch" it has error?

**Comment by xinntao** [comment]
1. In vanilla GAN, the mean of D outputs measures the realness or fakeness of the inputs. Thus, in theory, the D_fake and D_real should converge to both 0.5 (See https://arxiv.org/pdf/1406.2661.pdf  ). However, in practice, this is not the case. For example, you can find the D outputs for DCGAN in https://github.com/carpedm20/DCGAN-tensorflow, where D_real approaches to 1 and D_fake approaches to 0. 
Furthermore, the outputs of D are not necessarily the scores of realness or fakeness. 
In ESRGAN, we use Relativistic GAN, where the discriminator predict
relative realness instead of the absolute value. The meaning of D outputs is not clear. 

I attached one example for ESRGAN, D_real and D_fake are the values before sigmoid. 
![image](https://user-images.githubusercontent.com/17445847/49153330-66a7b400-f350-11e8-8a74-aeaa5a25b0e6.png)

2. The SRGAN model we provided is different from the official one. We also omit the BN layers and do some slight changes. But the main ideas -- using ResNet stucture, vgg+adv loss, are the same.

**Comment by Jing25** [comment]
Thank you for the explanation. It helps a lot.
